pkgs/development/libraries/mesa/default.nix

```
@@ -171,6 +171,8 @@ in stdenv.mkDerivation {
   # Needed to discover llvm-config for cross
   preConfigure = ''
     PATH=${llvmPackages.libllvm.dev}/bin:$PATH
+  '' + lib.optionalString (stdenv.targetPlatform.useLLVM or false) ''
+    export PKG_CONFIG_PATH_FOR_BUILD="${glslang}/lib/pkgconfig:$PKG_CONFIG_PATH_FOR_BUILD"
```

```suggestion
    export PKG_CONFIG_PATH_FOR_BUILD="${buildPackages.glslang}/lib/pkgconfig:$PKG_CONFIG_PATH_FOR_BUILD"
```

Otherwise it'll be wrong for cross



Why is this necessary when `depsBuildBuild` has `pkg-config` and `nativeBuildInputs` has `glslang` already.

---

I think the better question is why is our glslang propagating spirv-tools.

---

Wdym by propagating? It provides it. I just don't know why it's seeming to prefer the build native one.

---

It shouldn't do that, we have spirv-tools as a separate package.